docs/introduction/container-images.md

```diff
@@ -0,0 +1,253 @@
+
+Here! Each cgroup is separately mountable. 
+
+```bash
+ls -l /sys/fs/cgroup
+total 0
+dr-xr-xr-x 6 root root  0 Aug 29 15:00 blkio
+lrwxrwxrwx 1 root root 11 Aug 29 15:00 cpu -> cpu,cpuacct
+lrwxrwxrwx 1 root root 11 Aug 29 15:00 cpuacct -> cpu,cpuacct
+dr-xr-xr-x 6 root root  0 Aug 29 15:00 cpu,cpuacct
+dr-xr-xr-x 3 root root  0 Aug 29 15:00 cpuset
+dr-xr-xr-x 6 root root  0 Aug 29 15:00 devices
+dr-xr-xr-x 3 root root  0 Aug 29 15:00 freezer
+dr-xr-xr-x 3 root root  0 Aug 29 15:00 hugetlb
+dr-xr-xr-x 6 root root  0 Aug 29 15:00 memory
+lrwxrwxrwx 1 root root 16 Aug 29 15:00 net_cls -> net_cls,net_prio
+dr-xr-xr-x 3 root root  0 Aug 29 15:00 net_cls,net_prio
+lrwxrwxrwx 1 root root 16 Aug 29 15:00 net_prio -> net_cls,net_prio
+dr-xr-xr-x 3 root root  0 Aug 29 15:00 perf_event
+dr-xr-xr-x 6 root root  0 Aug 29 15:00 pids
+dr-xr-xr-x 2 root root  0 Aug 29 15:00 rdma
+dr-xr-xr-x 6 root root  0 Aug 29 15:00 systemd
+```
+
+Each of these is a folder, and the simplest one is "freezer." The way we control cgroups is via file system calls.
+So, to make a cgroup called "test" I create a directory called test.
+To move a process as a task into a cgroup, I can echo it there.
+
+## Containers
+
+Everything that claims to be a container technology is basically orchestrating 
+this cgroup and namespace system. You can draw boxes on your powerpoint, call containers
+"packages" or "encapsulated environments" or what not - a container technology is just
+some software to stick it's boogery fingers into these system APIs. Containers: 
+
+ - are encapsulations of an environment
+ - have their own process namespace
+ - the main container on the host is still associated with one process, and when you start that process the set within the container might kick off, and when you kill it the container's processes are also killed.
+
+
+### Container Namespaces
+
+Akin to the namespaces on your host, a container also has its own namespace? 
+Here it is helpful to think of a container like a little packaged
+environment, one that you can actually shell into like you would a server. When I shell
+inside the container I find that it has its own little process namespace.
+
+### Container Cgroups
+
+A cgroup is a "control group" and it's exactly that - a feature of the kernel
+that lets you allocate permissions for resources. So, for example, I might allocate
+a particular amount of memory, or CPU time, or network bandwidth to a set of tasks,
+these processes (point to processes) that are running on my computer. What else can we
+do? We can monitor them, reconfigure them, and even deny them access to resources.
+So these containers also have their own set of cgroups.
+
+## What about a container image?
+
+The words "container" and "image" are sometimes used synonymously, but are subtly two
+different things. The image is a binary sitting on your filesystem of actual stuff in
+some frozen state. In docker you start from a bunch of compressed archives put together,
```

Comment:
I think this is over complicating the matter at this point talking about docker and compressed archives.. Might just be sufficient to leave it with the 'Image = Template & Container = Instance of the template' description above.

Reply:
Agree.